### PR TITLE
yield setHook function

### PR DESCRIPTION
**Pull request by miguelcobain**
I faced a situation where I need to apply the `hook` modifier + some custom logic to the element.
The problem is that modifiers aren't composable in ember. So we can't create a custom modifier that applies another modifier.

For that reason, in this PR I'm yielding a `setHook` function, thus increasing the composability options for users.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
   },
   "devDependencies": {
     "@changesets/changelog-github": "^0.5.0",
-    "@changesets/cli": "^2.26.0"
+    "@changesets/cli": "^2.26.0",
+    "@glint/core": "^1.3.0"
```
Comment (miguelcobain) [inline package.json:21]: This is just to make vscode glint extension happy and correctly detect gts files in sub packages.

**Review by ynotdraw — APPROVED** [review]

**Comment by ynotdraw** [comment]
Thanks @miguelcobain! This is excellent.

**Comment by miguelcobain** [comment]
@ynotdraw I pushed a fix for the typecheck error CI was reporting. Please approve workflow. Thanks.

File: test-app/tests/integration/components/velcro-test.gts [hunk 2]
```diff
@@ -37,6 +38,35 @@ module('Integration | Component | velcro', function (hooks) {
...
+  test('it renders with setHook', async function (assert) {
+    let hookModifier = modifier((element: HTMLElement | SVGElement, [setHook]: [(element: HTMLElement | SVGElement) => void]) => {
+      setHook(element);
+    });
+
+
+    await render(<template>
+      <Velcro as |velcro|>
+        <div id="hook" {{hookModifier velcro.setHook}} style="width: 200px; height: 40px">
+          {{velcro.data.rects.reference.width}}
```
[the remark this answers is not in the document]
Comment (miguelcobain) [reply test-app/tests/integration/components/velcro-test.gts:50]: @nicolechung 
> Also the test doesn't show much other than it renders.

The test shows that it renders and that the content was correctly positioned and sized. Which means that the `setHook` works, just like the `hook` modifier.

> While the PR description is helpful, I can't quite see how another modifier would make use of this.

It would be a contrived example, but I'll explain my use case: Imagine you're writing a dropdown component witth ember-velcro. You want to yield a `trigger` modifier that does two things:
1. applies the hook from ember-velcro
2. attaches a click handler to toggle between the open/closed states

Without `setHook`, this would not be possible. With `setHook` however, we can pass that function to the modifier, and the modifier can call that function.

The dropdown component might look something like this:

```gjs
let myModifier = modifier((element, [setHook, handler]) => {
  // call ember-velcro's setHook
  setHook(element);
  
  // other custom logic
  element.addEventListener('click', handler);
  
  return () => {
    element.removeEventListener('click', handler);
  };
});

<template>
  <Velcro as |velcro|>
    {{yield (hash
      trigger=(modifier myModifier velcro.setHook onClick)
    }}
  </Velcro>
</template>
```

This is needed because there's no way in ember (that I know of) to combine two existing modifiers into a single one. If there was, this PR wouldn't be needed.

Being this such an edge case, I'm not sure if we should document it with this detail. It might distract users from the main usage. Maybe adding this in a [`<details>`](https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d) tag in markdown?

[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply test-app/tests/integration/components/velcro-test.gts:50]: I wouldn't consider this an edge case -- in design systems, this sort of problem is _IMMENSELY_ common.

[the remark this answers is not in the document]
Comment (miguelcobain) [reply test-app/tests/integration/components/velcro-test.gts:50]: @nicolechung documentation was added

File: README.md [hunk 3]
```diff
@@ -80,6 +80,46 @@ The `Velcro` component yields a single hash - 2 modifiers and 'velcro data':
...
+<details>
+ <summary>You can also use `velcro.setHook` yielded function for more complex composibility scenarios. Expand to read more.</summary>
+
+## `setHook`
+
+Imagine you're writing a dropdown component with ember-velcro. You want to yield a `trigger` modifier that does two things:
+1. sets an element as the "hook" for ember-velcro
+2. attaches a click handler to toggle between the open/closed states
+
+Without the yielded `setHook` function, this would not be possible. With `setHook` however, we can pass that function to the modifier, and the modifier can call that function with the element.
```
Comment (nicolechung) [inline README.md:92]: ✅  thank you! Very clear.

File: test-app/tests/integration/components/velcro-test.gts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolechung) [reply test-app/tests/integration/components/velcro-test.gts:50]: > The test shows that it renders and that the content was correctly positioned and sized. Which means that the setHook works, just like the hook modifier.

I think (hope?) that should be enough for the tests. What I meant is that a user can't tell why they'd need to use `setHook` by looking at the test alone (apologize, was multi-tasking a bit and forget to edit my comment).

The documentation looks terrific, thanks for adding that!

[the remark this answers is not in the document]
Comment (miguelcobain) [reply test-app/tests/integration/components/velcro-test.gts:50]: I like when tests self-document the usage of something. This test passes the setHook into a custom modifier like it is intended to be used.
I'll gladly accept suggestions on how to improve the test.
